packages/context/src/binding.ts

```diff
@@ -54,14 +71,19 @@ export class Binding {
 
   private readonly _key: string;
   private _tags: Set<string> = new Set();
+  private _scope: BindingScope;
+  private _cache: BoundValue;
+  private _getValue: (ctx?: Context) => BoundValue | Promise<BoundValue>;
```

Comment:
Making the `ctx` argument optional looks very suspicious to me. How do you expect bindings like `toClass` to resolve the value when the context argument is not provided (is `undefined`)?